peekaboo/server.py

```diff
@@ -435,70 +59,239 @@ def __init__(self, sock_file, job_queue, sample_factory, request_queue_size,
         @type request_queue_size: int
-        @param sock_group: An optional name of a group that shall own the
-                           socket.
-        @type sock_group: String
-        @param sock_mode: The permission bits of the socket.
-        @type sock_mode: Integer (bitmask)
         """
-        self.shutdown_requested = threading.Event()
-        self.shutdown_requested.clear()
-
         logger.debug('Starting up server.')
-        self.server = PeekabooStreamServer(
-            sock_file,
-            PeekabooStreamRequestHandler,
-            job_queue=job_queue,
-            sample_factory=sample_factory,
-            request_queue_size=request_queue_size)
+        self.app = sanic.Sanic("PeekabooAV", configure_logging=False)
+        self.app.config.FALLBACK_ERROR_FORMAT = "json"
+
+        # silence sanic to a reasonable amount
+        logging.getLogger('sanic.root').setLevel(logging.WARNING)
+        logging.getLogger('sanic.access').setLevel(logging.WARNING)
+
+        self.loop = asyncio.get_event_loop()
+        self.server_coroutine = self.app.create_server(
+            host=host, port=port, return_asyncio_server=True,
+            backlog=request_queue_size,
+            asyncio_server_kwargs=dict(start_serving=False))
+        self.server = None
+        self.job_queue = job_queue
+        self.sample_factory = sample_factory
+        self.db_con = db_con
+        # remember for diagnostics
+        self.host = host
+        self.port = port
+
+        self.app.add_route(self.hello, '/')
+        self.app.add_route(self.ping, '/ping')
+        self.app.add_route(self.scan, "/v1/scan", methods=['POST'])
+        self.app.add_route(
+            self.report, '/v1/report/<job_id:int>', methods=['GET'])
+
+    async def hello(self, _):
+        """ hello endpoint as fallback and catch all
+
+        @returns: hello world json response
+        """
+        return sanic.response.json({'hello': 'PeekabooAV'})
+
+    async def ping(self, _):
+        """ ping endpoint for diagnostics
+
+        @returns: pong json response
+        """
+        return sanic.response.json({'answer': 'pong'})
+
+    async def scan(self, request):
+        """ scan endpoint for job submission
+
+        @param request: sanic request object
+        @type request: sanic.Request
+        @returns: json response containing ID of newly created job
+        """
+        # this is sanic's multipart/form-data parser in a version that knows
+        # that our file field contains binary data. This allows transferring
+        # files without a filename. The generic parser would treat those as
+        # text fields and try to decode them using the form charset or UTF-8 as
+        # a fallback and cause errors such as: UnicodeDecodeError: 'utf-8'
+        # codec can't decode byte 0xc0 in position 1: invalid start byte
+        content_type = request.headers.getone(
+            'content-type', sanic.constants.DEFAULT_HTTP_CONTENT_TYPE
```

Comment:
Good catch! `DEFAULT_HTTP_CONTENT_TYPE` moved from `sanic.request` to `sanic.constants` between 20.12 and 21.3. It would still be available as `sanic.request.DEFAULT_HTTP_CONTENT_TYPE` but that seemed hackish to me. By coincidence I noticed that `request.content_type` does exactly the same thing, wraps these internals and is avaialable in all versions. So I switched to that which actually simplifies our code as well.

BTW: Rebased the whole branch to current master without further changes.